streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableAggregate.java

```diff
@@ -85,10 +85,19 @@ public void process(final K key, final Change<V> value) {
             final ValueAndTimestamp<T> oldAggAndTimestamp = store.get(key);
             final T oldAgg = getValueOrNull(oldAggAndTimestamp);
             final T intermediateAgg;
+            long newTimestamp = context().timestamp();
 
             // first try to remove the old value
             if (value.oldValue != null && oldAgg != null) {
                 intermediateAgg = remove.apply(key, value.oldValue, oldAgg);
+                // TODO: not sure if this make sense
+                // technically, if we remove a value from an aggregation,
```

Comment:
In current code, I think you are right -- however, I think it's better to keep both separate, as we might want to do an optimization to only send one `Change` record instead of two if the key for both is the same. Hence, I think the current code is more robust.

Reply:
> Is there a standard SP definition of aggregation result processing timestamp to look to?

Not really. The only definitions I am aware of, don't handle the delete case (most systems only support "fact streams" but no update semantics)

I guess that makes sense. I just wanted to have a discussion, because it's not completely straight forward. If we are all happy with taking "max" and to advance the timestamp, I have no objection at all.

Reply:
Sweet. Glad to have your expertise on this stuff!

Reply:
Max makes sense for me.